file-proxy/src/main/scala/hmda/uli/api/http/ProxyHttpApi.scala

```diff
@@ -0,0 +1,145 @@
+  }
+
+  val s3Settings = S3Settings(system)
+    .withBufferType(MemoryBufferType)
+    .withCredentialsProvider(awsCredentialsProvider)
+    .withS3RegionProvider(awsRegionProvider)
+    .withListBucketApiVersion(ListBucketVersion2)
+
+  def proxyHttpRoutes(oAuth2Authorization: OAuth2Authorization): Route = {
+    encodeResponse {
+      pathPrefix("file") {
+        //Modified Lar Route CSV
+        path("modifiedLar"/ "year" / Segment / "institution" / Segment / "csv") { (year, lei) =>
+          (extractUri & get) { uri =>
+            checkYearAvailable(dynamicYears, year) {
+              val s3Key = "prod/modified-lar/" + year + "/" + lei + ".csv"
+              streamingS3Route(s3Key)
+            }
+          }
+        } ~
+        //Modified Lar Header Route CSV
+        path("modifiedLar"/ "year" / Segment / "institution" / Segment / "csv" / "header") { (year, lei) =>
+          (extractUri & get) { uri =>
+            checkYearAvailable(dynamicYears, year) {
+              val s3Key = "prod/modified-lar/" + year + "/" + lei + "_header.csv"
+              streamingS3Route(s3Key)
+            }
+          }
+        } ~
+        //Modified Lar Route TXT
+        path("modifiedLar"/ "year" / Segment / "institution" / Segment / "txt") { (year, lei) =>
+          (extractUri & get) { uri =>
+            checkYearAvailable(dynamicYears, year) {
+              val s3Key = "prod/modified-lar/" + year + "/" + lei + ".txt"
+              streamingS3Route(s3Key)
+            }
+          }
+        } ~
+        //Modified Lar Header Route TXT
+        path("modifiedLar"/ "year" / Segment / "institution" / Segment / "txt" / "header") { (year, lei) =>
+          (extractUri & get) { uri =>
+            checkYearAvailable(dynamicYears, year) {
+              val s3Key = "prod/modified-lar/" + year + "/" + lei + "_header.txt"
+              streamingS3Route(s3Key)
+            }
+          }
+        } ~
+        //IRS Report Route
+        path("reports" / "irs" / "year" / Segment / "institution" / Segment) { (year, lei) =>
+          (extractUri & get) { uri =>
+            oAuth2Authorization.authorizeTokenWithLeiOrRole(lei, hmdaAdminRole) { _ =>
+              checkYearAvailable(irsYears, year) {
+                val s3Key = "prod/reports/disclosure/" + year + "/" + lei + "/nationwide/IRS.csv"
+                streamingS3Route(s3Key)
+              }
+            }
+          }
+        }
+      }
+    } 
+  }
+
+  private def retrieveData(path: String): Future[Option[Source[ByteString, NotUsed]]] = {
+    val timeout: Timeout = Timeout(config.getInt("hmda.http.timeout").seconds)
+    S3.download("cfpb-hmda-public", path).withAttributes(S3Attributes.settings(s3Settings)).runWith(Sink.head)
+      .map(opt => opt.map { case (source, _) => source })
+  }
+
+  private def streamingS3Route(s3Key: String): Route = {
+    val fStream: Future[Source[ByteString, NotUsed]] = retrieveData(s3Key).flatMap {
```

Comment:
A map without later creating a Future should suffice here. Something we can note in a future refactor